test/integration/targets/ini_file/tasks/main.yml

```diff
@@ -121,3 +121,92 @@
       - result5.changed == True
       - result5.msg == 'section removed'
       - content5 == ""
+
+- name: Ensure "hate=coke" is created within no section
+  ini_file:
+    section:
+    path: "{{ output_file }}"
+    option: hate
+    value: coke
+  register: result6
+
+- name: set expected content and get current ini file content
+  set_fact:
+    expected6: "hate = coke"
+    content6: "{{ lookup('file', output_file) }}"
+
+- name: assert 'changed' is true and content is OK (no section)
+  assert:
+    that:
+      - result6|changed
+      - result6.msg == 'option added'
+      - content6 == expected6
+
+- name: Ensure "hate=coke" is modified as "hate=water" within no section
+  ini_file:
+    path: "{{ output_file }}"
+    option: hate
+    value: water
+    section:
+  register: result7
+
+- name: set expected content and get current ini file content
+  set_fact:
+    expected7: "hate = water"
+
+    content7: "{{ lookup('file', output_file) }}"
+
+- name: assert 'changed' is true and content is OK (no section)
+  assert:
+    that:
+      - result7.changed == True
+      - result7.msg == 'option changed'
+      - content7 == expected7
+
+- name: remove option 'hate' within no section
+  ini_file:
+    section:
+    path: "{{ output_file }}"
+    option: hate
+    state: absent
+  register: result8
+
+- name: get current ini file content
+  set_fact:
+    content8: "{{ lookup('file', output_file) }}"
+
+- name: assert changed (no section)
+  assert:
+    that:
+      - result8.changed == True
+      - result8.msg == 'option changed'
+      - content8 == ""
+
+- name: Check add option without section before existing section
+  block:
+    - name: Add option with section
+      ini_file:
+        path: "{{ output_file }}"
+        section: drinks
+        option: hate
+        value: water
+    - name: Add option without section
+      ini_file:
+        path: "{{ output_file }}"
+        section:
+        option: like
+        value: beer
```

Comment:
please remove all references to alcoholic beverages; I'm tired of being told by inhabitants of (in particular) north american countries that we use too much alcohol in examples. #sigh